### ISSUE TITLE
Beancount LSP reports spurious false errors in multi-file projects unless top-level file is open

### ISSUE THREAD

**Issue by jansol** [zed-industries/zed#21586]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I suspect this is a similar case to #21547

I have a beancount setup where I have a "main" file that defines currencies etc and then includes files from a subdirectory with a wildcard for the actual transactions. This means that I generally just need to open the latest transactions file to append new data. However the LSP is having none of it and flags every non-empty, non-comment line of the transactions file as having an error *unless* I open the main file in another tab, *save it once* (to "touch" it on disk) and keep it open. Closing the main file and saving the transactions file makes the errors reappear.

### Environment

N/A

**Comment by jansol** [comment]
No, neovim complains about everything when opening a transaction file directly as well.

**Comment by notpeter** [comment]
I think the correct behavior here is for the language server to scan the necessary files in the workspace directory and include all of them when calculating diagnostics.  This is what rust-analyzer and some other well behaved LSPs do.

I'm not sure how Zed would even work around this other than to tell the LSP we've opened **all** files in a given project directory, but we couldn't possibly know *which* files would be relevant up front as they would be language specific. 

As a result I consider this an upstream LSP bug and am closing as unactionable.
